rpc/client.ts

```diff
@@ -102,10 +102,15 @@ export class Client<
         )
       },
     }) as ClientSubscribeListener<SendErrorData, HandlerErrorData>
-    this.pendingSubscriptions[message.id] = listenerBound
+    this.pendingSubscriptions[message.id] = (maybeError) => {
+      listenerBound(maybeError)
+      if (maybeError instanceof Error) {
+        stop()
+      }
+    }
     this.call(message)
       .then((maybeError) => {
-        if (maybeError instanceof Error) {
+        if (maybeError instanceof Error || maybeError.error) {
```

Comment:
bug fix, if the subscription creation is not successful it should stop the subscription